src/main/java/BasicMAPF/Solvers/PrioritisedPlanning/PrioritisedPlanning_Solver.java

```diff
@@ -157,7 +155,15 @@ protected void init(MAPF_Instance instance, RunParameters parameters){
             this.agents.sort(this.agentComparator);
         }
         // heuristic
-        this.singleAgentGAndH = Objects.requireNonNullElseGet(parameters.singleAgentGAndH, () -> new DistanceTableSingleAgentHeuristic(this.agents, instance.map));
+//        this.singleAgentGAndH = Objects.requireNonNullElseGet(parameters.singleAgentGAndH, () -> new DistanceTableSingleAgentHeuristic(this.agents, instance.map));
+
+        this.singleAgentGAndH = Objects.requireNonNullElseGet(
+                parameters.singleAgentGAndH,
+                () -> this.transientMAPFBehaviour == TransientMAPFBehaviour.transientMAPFsstWithBlacklist ?
+                        new ServiceTimeGAndH(instance.agents, instance.map, null) :
+                        new DistanceTableSingleAgentHeuristic(this.agents, instance.map)
+        );
+
         if (this.singleAgentGAndH instanceof CachingDistanceTableHeuristic){
             ((CachingDistanceTableHeuristic)this.singleAgentGAndH).setCurrentMap(instance.map);
         }
```

Comment:
oh this is actually a little tricky because we will miss this check... I think it will still work fine even in lifelong because this is just extra, but I guess I'll improve this interface after we merge